src/plugins/url_forwarding/public/forward_app/forward_app.ts

```diff
@@ -23,23 +23,16 @@ export const createLegacyUrlForwardApp = (
   async mount(params: AppMountParameters) {
     const hash = params.history.location.hash.substr(1);
 
-    if (!hash) {
-      const [, , kibanaLegacyStart] = await core.getStartServices();
-      kibanaLegacyStart.navigateToDefaultApp();
-    }
-
     const [
       {
         application,
         http: { basePath },
       },
     ] = await core.getStartServices();
 
-    const result = await navigateToLegacyKibanaUrl(hash, forwards, basePath, application);
-
-    if (!result.navigated) {
-      const [, , kibanaLegacyStart] = await core.getStartServices();
-      kibanaLegacyStart.navigateToDefaultApp();
+    const { navigated } = navigateToLegacyKibanaUrl(hash, forwards, basePath, application);
+    if (!navigated) {
```

Comment:
:information_source: This code is required to make sure that the following URLs keep working: `/app/kibana#/foobarOrSomeOtherNonExistingNonesense` Earlier the code navigated to the configured `defaultAppId`. With this change we'll just navigate to `/` in the current space (thus the `basePath.get()`), which will itself triggering the behavior looking for the configured `defaultRoute` and bringing us there.